src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -1014,10 +1014,18 @@ public Path zipRepository(Repository repository, String zipFilename, String repo
             zipFilenameWithoutSlash += ".zip";
         }
 
-        Path zipFilePath = Paths.get(repositoryDir, "zippedRepos", zipFilenameWithoutSlash);
-        Files.createDirectories(Paths.get(repositoryDir, "zippedRepos"));
-
-        return zipFileService.createZipFileWithFolderContent(zipFilePath, repository.getLocalPath());
+        Path zipFilePath = Paths.get(repositoryDir, zipFilenameWithoutSlash);
+        Files.createDirectories(Paths.get(repositoryDir));
+        zipFileService.createZipFileWithFolderContent(zipFilePath, repository.getLocalPath());
+
+        // Schedule created files and directories for deletion
+        fileService.scheduleForDeletion(zipFilePath, 5);
+        fileService.scheduleForDeletion(repository.getLocalPath().getParent(), 5);
+        fileService.scheduleForDeletion(zipFilePath.getParent(), 6);
+        // Delete the zipped repository
+        deleteLocalRepository(repository);
+
+        return zipFilePath;
```

Comment:
Try to avoid deleting repositories here. Create a a single parent directory where you store the repos. Then you can simply delete it after you zipped its contents